File: leetcode/200/LC102.go
func levelOrder(root *TreeNode) [][]int {
	ret := [][]int{}
	if root == nil{
		return ret;
	}

	layer := []*TreeNode{root}
	for i:=0; len(layer) > 0; i++{
		ret = append(ret, []int{})

		cnt := len(layer)

		newLayer := []*TreeNode{}
		for j := 0; j < cnt; j++{
			node := layer[j]
			ret[i] = append(ret[i],node.Val)

			if node.Left != nil{
				newLayer = append(newLayer,node.Left)
			}
			if node.Right != nil{
				newLayer = append(newLayer,node.Right)
			}
		}

		layer = newLayer
	}

	return ret
}

File: leetcode/100/LC102.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;

public class LC102 {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> ret = new ArrayList<>();
        if (root == null) {
            return ret;
        }
        List<TreeNode> layer = new LinkedList<>();
        layer.add(root);

        while (!layer.isEmpty()) {
            List<Integer> tmp = new ArrayList<>();
            int cnt = layer.size();
            for(int i = 0; i < cnt; i++) {
                TreeNode node = layer.removeFirst();
                tmp.add(node.val);
                if (node.left != null) {
                    layer.add(node.left);
                }
                if (node.right != null) {
                    layer.add(node.right);
                }
            }
            ret.add(tmp);
        }
        return ret;
    }

    class TreeNode {
         int val;
         TreeNode left;
         TreeNode right;
         TreeNode() {}
         TreeNode(int val) { this.val = val; }
         TreeNode(int val, TreeNode left, TreeNode right) {
             this.val = val;
             this.left = left;
             this.right = right;
         }
     }
}